### PR TITLE
fixes py2-h5py dependencies

### PR DESCRIPTION
**Pull request by iahmad-khan**

### CODE REVIEW

**Comment by iahmad-khan** [comment]
please test

**Comment by iahmad-khan** [comment]
@davidlt this should fix the py2-h5py dependency issues.

File: cython.spec [hunk 1]
```diff
@@ -21,5 +21,3 @@ sed -ideleteme 's|#!.*/bin/python|#!/usr/bin/env python|' \
   %{i}/${PYTHON_LIB_SITE_PACKAGES}/Cython/Debugger/libpython.py
 
 find %{i} -name '*deleteme' -delete
-
-find %{i}/${PYTHON_LIB_SITE_PACKAGES} -name '*.egg-info' -print0 | xargs -0 rm -f
```
Comment (davidlt) [inline cython.spec:25]: Why this is removed? Long-term policy was that we do not provide _egg_ stuff.

File: py2-h5py.spec [hunk 2]
```diff
@@ -2,8 +2,8 @@
 ## INITENV +PATH PYTHONPATH %{i}/$PYTHON_LIB_SITE_PACKAGES
 %define my_name %(echo %n | cut -f2 -d-)
 Source: https://github.com/%{my_name}/%{my_name}/archive/%{realversion}.tar.gz
-Requires: python py2-numpy hdf5
-BuildRequires: py2-setuptools hdf5
+Requires: python py2-numpy hdf5 py2-six
+BuildRequires: py2-setuptools cython py2-pkgconfig
```
Comment (davidlt) [inline py2-h5py.spec:6]: Have you looked into how pkgconfig is being used? We need to ensure it does not go to look into the system.

File: py2-six.spec [hunk 3]
```diff
@@ -13,4 +13,3 @@ python setup.py build
 
 %install
 python setup.py install --single-version-externally-managed --record=/dev/null --skip-build --prefix=%{i}
-find %{i}/${PYTHON_LIB_SITE_PACKAGES} -name '*.egg-info' -print0 | xargs -0 rm -rf
```
Comment (davidlt) [inline py2-six.spec:16]: Again, way we keeping the eggs?

File: cython.spec (hunk 1 above)
Comment (iahmad-khan) [reply cython.spec:25]: actually the h5py setup has no way to find our cython without this.

File: py2-six.spec (hunk 3 above)
Comment (iahmad-khan) [reply py2-six.spec:16]: same reason as above

File: py2-h5py.spec (hunk 2 above)
Comment (iahmad-khan) [reply py2-h5py.spec:6]: yes

Comment (smuzaffar) [reply py2-h5py.spec:6]: I think pkgconfig is only needed if you call setup.py test which we are not calling at build time (may be we should)

**Pull request merged**